utils/swift_build_support/swift_build_support/products/swiftsystem.py

```diff
@@ -0,0 +1,140 @@
+        return True
+
+
+    # TODO: I copied the below defs from swift-testing, what should I do here?
+    def should_clean(self, host_target):
+        # Workaround for 'swift-testing' not detecting compiler/stdlib changes.
+        return True
+    def should_build(self, host_target):
+        return True
+    def should_test(self, host_target):
+        # TODO: Implement.
+        return False
+
+    def should_install(self, host_target):
+        return True # WAS: self.args.install_swift_system
+
+    def _cmake_product(self, host_target):
+        build_root = os.path.dirname(self.build_dir)
+        build_dir = os.path.join(
+            build_root, '%s-%s' % (self.product_name(), host_target))
+
+        return SwiftSystemCMakeShim(
+            args=self.args,
+            toolchain=self.toolchain,
+            source_dir=self.source_dir,
+            build_dir=build_dir)
+
+    def _for_each_host_target(self, base_target, body):
+        body(base_target)
+
+        # TODO: what do we do here? We're not built on Darwin
+
+        # For Darwin host, 'build' is only called for the builder.
+        # Manually iterate the cross compile hosts.
+        if self.has_cross_compile_hosts() and self.is_darwin_host(base_target):
+            for target in self.args.cross_compile_hosts:
+                body(target)
+
+    def _clean_with_cmake(self, host_target):
+        self._cmake_product(host_target).clean(host_target)
+
+    def clean(self, host_target):
+        self._for_each_host_target(host_target, self._clean_with_cmake)
+
+    def _build_with_cmake(self, host_target):
+        self._cmake_product(host_target).build(host_target)
+
+    def build(self, host_target):
+        self._for_each_host_target(host_target, self._build_with_cmake)
+
+    def _install_with_cmake(self, host_target):
+        self._cmake_product(host_target).install(host_target)
+
+    def install(self, host_target):
+        self._for_each_host_target(host_target, self._install_with_cmake)
+
+
+class SwiftSystemCMakeShim(cmake_product.CMakeProduct):
+    def clean(self, host_target):
+        shell.rmtree(self.build_dir)
+
+    def build(self, host_target):
+        override_deployment_version = None
+
+        # TODO: Do we need this, or should we adjust our own version?
+        if host_target.startswith('macosx'):
+            if Version(self.args.darwin_deployment_version_osx) < Version('10.15'):
+                override_deployment_version = '10.15'
+
+        build_shared_libs = not host_target.startswith('wasi')
+        self.cmake_options.define('BUILD_SHARED_LIBS',
+                                  'TRUE' if build_shared_libs else 'FALSE')
+
+        # Use empty CMake install prefix, since the `DESTDIR` env var is set by
+        # `install_with_cmake` later which already has the same prefix.
+        self.cmake_options.define('CMAKE_INSTALL_PREFIX', '')
+
+        self.cmake_options.define('CMAKE_BUILD_TYPE', self.args.build_variant)
```

Comment:
This is the default build variant, you'll want to override it with the system_build_variant flag you are adding.